src/dialogs/more-info/ha-more-info-history.ts

```diff
@@ -83,19 +92,43 @@ export class MoreInfoHistory extends LitElement {
     super.updated(changedProps);
     if (!this.entityId) {
       clearInterval(this._historyRefreshInterval);
+      this._stateHistory = undefined;
+      this._entries = undefined;
+      this._lastLogbookDate = undefined;
+      this._throttleHistoryFunction = undefined;
+      return;
     }
 
     if (changedProps.has("entityId")) {
       this._stateHistory = undefined;
       this._entries = undefined;
+      this._lastLogbookDate = undefined;
+      this._throttleHistoryFunction = undefined;
 
       this._getStateHistory();
       this._getLogBookData();
 
-      clearInterval(this._historyRefreshInterval);
-      this._historyRefreshInterval = window.setInterval(() => {
-        this._getStateHistory();
-      }, 60 * 1000);
+      this._throttleHistoryFunction = throttleAndQueue(
+        () => {
+          this._getStateHistory();
+          this._getLogBookData();
+        },
+        10000,
+        5000
+      );
+      return;
+    }
```

Comment:
Add a throttle...

It also has a 5 second delay on the first call because I was seeing if I immediately called the API the state change would be in the return...

This is probably not the best way. Open to suggestions

Reply:
I don't understand the 5 second delay?

Reply:
Oh is see what you mean now, the history and logbook requests don't give the data back...